### PR TITLE
Add table aws_securityhub_standards_subscription. Closes #277

### LINKED ISSUE #277: Add table aws_securityhub_standards_subscription
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.38.10/service/securityhub#StandardsSubscription

### PR DESCRIPTION
**Pull request by rajeshbal65**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_securityhub_standards_subscription []

PRETEST: tests/aws_securityhub_standards_subscription

TEST: tests/aws_securityhub_standards_subscription
Running terraform
aws_securityhub_account.named_test_resource: Creating...
aws_securityhub_account.named_test_resource: Creation complete after 4s [id=352685002396]
aws_securityhub_standards_subscription.cis: Creating...
aws_securityhub_standards_subscription.pci_321: Creating...
aws_securityhub_standards_subscription.pci_321: Creation complete after 3s [id=arn:aws:securityhub:us-east-1:352685002396:subscription/pci-dss/v/3.2.1]
aws_securityhub_standards_subscription.cis: Creation complete after 7s [id=arn:aws:securityhub:us-east-1:352685002396:subscription/cis-aws-foundations-benchmark/v/1.2.0]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 3 added, 0 changed, 0 destroyed.

Outputs:

account_id = "352685002396"
aws_partition = "aws"
aws_region = "us-east-1"
cis_arn = "arn:aws:securityhub:us-east-1:352685002396:subscription/cis-aws-foundations-benchmark/v/1.2.0"
pci_321_arn = "arn:aws:securityhub:us-east-1:352685002396:subscription/pci-dss/v/3.2.1"

Running SQL query: test-get-query.sql
[
  {
    "enabled_by_default": true,
    "name": "CIS AWS Foundations Benchmark v1.2.0",
    "standards_arn": "arn:aws:securityhub:::ruleset/cis-aws-foundations-benchmark/v/1.2.0",
    "standards_subscription_arn": "arn:aws:securityhub:us-east-1:352685002396:subscription/cis-aws-foundations-benchmark/v/1.2.0"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "standards_input": {},
    "standards_status": "INCOMPLETE",
    "standards_subscription_arn": "arn:aws:securityhub:us-east-1:352685002396:subscription/cis-aws-foundations-benchmark/v/1.2.0"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "enabled_by_default": true,
    "name": "CIS AWS Foundations Benchmark v1.2.0",
    "standards_arn": "arn:aws:securityhub:::ruleset/cis-aws-foundations-benchmark/v/1.2.0",
    "standards_status": "INCOMPLETE",
    "standards_subscription_arn": "arn:aws:securityhub:us-east-1:352685002396:subscription/cis-aws-foundations-benchmark/v/1.2.0"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:securityhub:us-east-1::standards/pci-dss/v/3.2.1"
    ],
    "name": "PCI DSS v3.2.1",
    "region": "us-east-1",
    "title": "PCI DSS v3.2.1"
  }
]
✔ PASSED

POSTTEST: tests/aws_securityhub_standards_subscription

TEARDOWN: tests/aws_securityhub_standards_subscription

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  name,
  standards_arn description,
  region
from
  aws_securityhub_standards_subscription;

+-------------------------------------------------+-------------------------------------------------------------------------------------------+-----------+
| name                                            | description                                                                               | region    |
+-------------------------------------------------+-------------------------------------------------------------------------------------------+-----------+
| PCI DSS v3.2.1                                  | arn:aws:securityhub:us-east-1::standards/pci-dss/v/3.2.1                                  | us-east-1 |
| CIS AWS Foundations Benchmark v1.2.0            | arn:aws:securityhub:::ruleset/cis-aws-foundations-benchmark/v/1.2.0                       | us-east-1 |
| AWS Foundational Security Best Practices v1.0.0 | arn:aws:securityhub:us-east-1::standards/aws-foundational-security-best-practices/v/1.0.0 | us-east-1 |
+-------------------------------------------------+-------------------------------------------------------------------------------------------+-----------+
```

### List enabled standards in security hub

```sql
select
  name,
  standards_arn,
  enabled_by_default
from
  aws_securityhub_standards_subscription
where
  enabled_by_default;

+-------------------------------------------------+-------------------------------------------------------------------------------------------+--------------------+
| name                                            | standards_arn                                                                             | enabled_by_default |
+-------------------------------------------------+-------------------------------------------------------------------------------------------+--------------------+
| CIS AWS Foundations Benchmark v1.2.0            | arn:aws:securityhub:::ruleset/cis-aws-foundations-benchmark/v/1.2.0                       | true               |
| AWS Foundational Security Best Practices v1.0.0 | arn:aws:securityhub:us-east-1::standards/aws-foundational-security-best-practices/v/1.0.0 | true               |
+-------------------------------------------------+-------------------------------------------------------------------------------------------+--------------------+
```

### List standards where standard status is not ready

```sql
select
  name,
  standards_arn,
  standards_subscription_arn,
  standards_status
from
  aws_securityhub_standards_subscription
where
  standards_status <> 'READY';

+-------------------------------------------------+-------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------+------------------+
| name                                            | standards_arn                                                                             | standards_subscription_arn                                                                               | standards_status |
+-------------------------------------------------+-------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------+------------------+
| AWS Foundational Security Best Practices v1.0.0 | arn:aws:securityhub:us-east-1::standards/aws-foundational-security-best-practices/v/1.0.0 | arn:aws:securityhub:us-east-1:352685002396:subscription/aws-foundational-security-best-practices/v/1.0.0 | INCOMPLETE       |
| CIS AWS Foundations Benchmark v1.2.0            | arn:aws:securityhub:::ruleset/cis-aws-foundations-benchmark/v/1.2.0                       | arn:aws:securityhub:us-east-1:352685002396:subscription/cis-aws-foundations-benchmark/v/1.2.0            | INCOMPLETE       |
+-------------------------------------------------+-------------------------------------------------------------------------------------------+----------------------------------------------------------------------------------------------------------+------------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_securityhub_standards_subscription.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/service/securityhub"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
```
Comment (khushboo9024) [inline aws/table_aws_securityhub_standards_subscription.go:11]: //// TABLE DEFINITION

File: aws/table_aws_securityhub_standards_subscription.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+func tableAwsSecurityHubStandardSubscription(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_securityhub_standards_subscription",
+		Description: "AWS Securityhub standards",
```
Comment (khushboo9024) [inline aws/table_aws_securityhub_standards_subscription.go:15]:
```suggestion
		Description: "AWS Security Hub Standards Subscription",
```

File: aws/table_aws_securityhub_standards_subscription.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("StandardsArn").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listSecurityHubStandardSubcriptions(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```
Comment (khushboo9024) [inline aws/table_aws_securityhub_standards_subscription.go:85]:
```suggestion
func listSecurityHubStandardsSubcriptions(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```

File: aws/table_aws_securityhub_standards_subscription.go [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_securityhub_standards_subscription.go:87]:
```suggestion
```

File: aws/table_aws_securityhub_standards_subscription.go [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+	}
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getSecurityHubStandardSubcription(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getSecurityHubStandardSubcription")
+
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_securityhub_standards_subscription.go:118]:
```suggestion
```

File: aws/table_aws_securityhub_standards_subscription.go [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+	}
+	for _, standards := range resp.Standards {
+		d.StreamListItem(ctx, standards)
+	}
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getSecurityHubStandardSubcription(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```
Comment (khushboo9024) [inline aws/table_aws_securityhub_standards_subscription.go:115]:
```suggestion
func getSecurityHubStandardsSubcription(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```
change wherever required.

File: aws/table_aws_securityhub_standards_subscription.go [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+	plugin.Logger(ctx).Trace("listSecurityHubStandardSubcriptions", "AWS_REGION", region)
+
+	// Create session
+	svc, err := SecurityHubService(ctx, d, region)
+	if err != nil {
+		return nil, err
+	}
+
+	// List call
+	resp, err := svc.DescribeStandards(&securityhub.DescribeStandardsInput{})
```
Comment (khushboo9024) [inline aws/table_aws_securityhub_standards_subscription.go:102]: Can we use `DescribeStandardsPages` api for list call ?

File: aws/table_aws_securityhub_standards_subscription.go [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+	input := &securityhub.DescribeStandardsInput{}
+
+	// Get call
+	standards, err := svc.DescribeStandards(input)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, item := range standards.Standards {
+		if *item.Name == name {
```
Comment (khushboo9024) [inline aws/table_aws_securityhub_standards_subscription.go:142]: This is not a get call , here we are just filtering list call.
Do we need this function in this case?

File: aws/table_aws_securityhub_standards_subscription.go [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+			return item, nil
+		}
+	}
+	return nil, nil
+}
+
+func GetEnabledStandards(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("GetEnabledStandards")
+
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_securityhub_standards_subscription.go:152]:
```suggestion
```

File: docs/tables/aws_securityhub_standards_subscription.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+Security Hub also generates its own findings as the result of running automated and continuous checks against the rules in a set of supported security standards. These checks provide a readiness score and identify specific accounts and resources that require attention.
+
+## Examples
+
+### Basic info
+
+```sql
+select
+  name,
+  standards_arn description,
```
Comment (khushboo9024) [inline docs/tables/aws_securityhub_standards_subscription.md:12]: syntax is wrong

File: docs/tables/aws_securityhub_standards_subscription.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+  region
+from
+  aws_securityhub_standards_subscription;
+```
+
+
+### List enabled standards in security hub
```
Comment (khushboo9024) [inline docs/tables/aws_securityhub_standards_subscription.md:19]:
```suggestion
### List enabled security hub standards
```

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Please make the suggested changes and resolve conflict.

File: aws-test/tests/aws_securityhub_standards_subscription/test-hydrate-expected.json [hunk 12]
```diff
@@ -0,0 +1,7 @@
+[
+  {
+    "standards_input": {},
+    "standards_status": "INCOMPLETE",
+    "standards_subscription_arn": "arn:aws:securityhub:us-east-1:352685002396:subscription/cis-aws-foundations-benchmark/v/1.2.0"
```
Comment (khushboo9024) [inline aws-test/tests/aws_securityhub_standards_subscription/test-hydrate-expected.json:5]: why we are hardcoding account id, region and partition ?

File: aws-test/tests/aws_securityhub_standards_subscription/test-turbot-expected.json [hunk 13]
```diff
@@ -0,0 +1,10 @@
+[
+  {
+    "akas": [
+      "arn:aws:securityhub:us-east-1::standards/pci-dss/v/3.2.1"
+    ],
+    "name": "PCI DSS v3.2.1",
+    "region": "us-east-1",
```
Comment (khushboo9024) [inline aws-test/tests/aws_securityhub_standards_subscription/test-turbot-expected.json:7]: why we are hardcoding region ?

File: aws/table_aws_securityhub_standards_subscription.go [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+
+	"github.com/aws/aws-sdk-go/service/securityhub"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsSecurityHubStandardSubscription(_ context.Context) *plugin.Table {
```
Comment (khushboo9024) [inline aws/table_aws_securityhub_standards_subscription.go:14]:
```suggestion
func tableAwsSecurityHubStandardsSubscription(_ context.Context) *plugin.Table {
```

File: aws/table_aws_securityhub_standards_subscription.go [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+	return &plugin.Table{
+		Name:        "aws_securityhub_standards_subscription",
+		Description: "Security Hub Standards Subscription",
```
Comment (khushboo9024) [inline aws/table_aws_securityhub_standards_subscription.go:17]:
```suggestion
		Description: "AWS Security Hub Standards Subscription",
```

File: docs/tables/aws_securityhub_standards_subscription.md [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+
+## Examples
+
+### Basic info
+
+```sql
+select
+  name,
+  standards_arn,
+  description region
```
Comment (khushboo9024) [inline docs/tables/aws_securityhub_standards_subscription.md:13]: Syntax is wrong.

File: aws/table_aws_securityhub_standards_subscription.go (hunk 14 above)
Comment (khushboo9024) [reply aws/table_aws_securityhub_standards_subscription.go:15]: Please change this function name.

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Please change the function name from `tableAwsSecurityHubStandardSubscription` to `tableAwsSecurityHubStandardsSubscription`

**Review by khushboo9024 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Comment by cbruno10** [comment]
@rajeshbal65 Is this table combining security hub standard info, and standard subscription info? How are standards and standard subscriptions related to one another? Can standards exist by themselves, and if so, do they have valuable info in them? Also, how does standards controls relate to standards?

**Comment by rajeshbal65** [comment]
@cbruno10 Yes this table is combination security hub standard, and standard subscriptions. Standards gives the available  standards information and Standard subscription gives the whether security hub is subscribed to the standards and its status. 

Standards can exist by themselves but don’t have any valuable information in them.

**Comment by rajlearner17** [comment]
@cbruno10 can you pls check this to close

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Pull request merged**